backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -312,6 +313,8 @@ IGitMacheteRepositorySnapshot createSnapshot(IBranchLayout branchLayout) throws
       var rootBranchCreationResults = Try.sequence(rootBranchTries).getOrElseThrow(GitMacheteException::getOrWrap).toList();
       var rootBranches = rootBranchCreationResults.flatMap(creationResult -> creationResult.getCreatedBranches());
       var skippedBranchNames = rootBranchCreationResults.flatMap(creationResult -> creationResult.getSkippedBranchNames());
+      var duplicatedBranchNames = rootBranchCreationResults
+          .flatMap(creationResult -> creationResult.getDuplicatedBranchNames()).toSet().toList();
```

Comment:
Hmm why does have it be both `toSet` and `toList`?  `toSet` for removing duplicates from the list of duplicates itself? ;)

Reply:
:yes:

Reply:
This actually suggests that `duplicatedBranchNames` (and possibly `skippedBranchNames`) could instead be represented as `Set` everywhere (since we DO care about things being unique in these collections, but DON'T care about ordering &mdash; at least not until we show them to the user, when we might want to sort them first)